src/styleguide/renderer/react/preview.tsx

```diff
@@ -1,321 +0,0 @@
-	}
-
-	public componentDidCatch(error: Error): void {
-		this.setState({ errorMessage: error.message });
-	}
-
-	public render(): React.ReactNode {
-		if (this.state.errorMessage) {
-			return (
-				<ErrorMessage
-					patternName={this.props.element.getName()}
-					error={this.state.errorMessage}
-				/>
-			);
-		} else {
-			return <div data-sketch-name={this.props.element.getName()}>{this.props.children}</div>;
-		}
-	}
-}
-
-@observer
-class Preview extends React.Component<PreviewProps> {
-	@MobX.observable protected highlightArea: HighlightArea;
-	protected patternWrapperRef: PatternWrapper;
-
-	public constructor(props: PreviewProps) {
-		super(props);
-		this.highlightArea = new HighlightArea();
-	}
-
-	// tslint:disable-next-line:no-any
-	protected collectChildren(componentProps: any, pageElement: PageElement): void {
-		const pattern = pageElement.getPattern();
-
-		if (pattern) {
-			pattern.getSlots().forEach(slot => {
-				const slotId = slot.getId();
-				const reactPropertyName = slotId === Pattern.DEFAULT_SLOT_ID ? 'children' : slotId;
-				const slotContents = pageElement.getSlotContents(slotId);
-				componentProps[reactPropertyName] = slotContents.map((child, index) =>
-					this.createComponent(child)
-				);
-			});
-		}
-	}
-
-	// tslint:disable-next-line:no-any
-	protected collectPropertyValues(componentProps: any, pageElement: PageElement): void {
-		const pattern = pageElement.getPattern() as Pattern;
-		pattern.getProperties().forEach(property => {
-			const propertyId = property.getId();
-			const value = pageElement.getPropertyValue(propertyId);
-
-			componentProps[propertyId] = property.convertToRender(value);
-		});
-	}
-
-	public componentDidMount(): void {
-		this.triggerHighlight();
-	}
-
-	public componentDidUpdate(prevProps: PreviewProps): void {
-		this.triggerHighlight();
-	}
-
-	protected createAssetComponent(pageElement: PageElement): JSX.Element {
-		const src = pageElement.getPropertyValue(AssetProperty.SYNTHETIC_ASSET_ID) as string;
-		return this.createWrapper(pageElement, <Placeholder src={src} />);
-	}
-
-	/**
-	 * Converts a JSON-serializable declaration of a pattern, primitive, or collection
-	 * into a React component (or primitive), deep-traversing through properties and children.
-	 * @param value The value, may be a page element (a pattern declaration),
-	 * a primitive like a string, number, boolean, null, etc.,
-	 * or an array or object of such values.
-	 * @returns A React component in case of a page element, the primitive in case of a primitive,
-	 * or an array or object with values converted in the same manner, if an array resp. object is provided.
-	 */
-	protected createComponent(pageElement: PageElement): React.ReactNode {
-		const pattern = pageElement.getPattern();
-		if (!pattern) {
-			return null;
-		}
-
-		try {
-			const patternId: string = pattern.getId();
-			if (patternId === Pattern.SYNTHETIC_TEXT_ID) {
-				return this.createStringComponent(pageElement);
```

Comment:
The string and asset synthetic components got lost on the way.
The Placeholder was a separate tsx file, while string was just... as string.
Both feature no longer work in the new solution.